.gitmodules

```diff
@@ -1,3 +1,3 @@
-[submodule ".devcontainer"]
-	path = .devcontainer
+[submodule ".github/devcontainer"]
```

Comment:
Somehow makes me _slightly_ nervous to use `.github` for this, given how heavily rely on GHA workflows. Is the convention to use this dir for this, and with submodule?

Reply:
.github dir is a general purpose directory. all workflows files should be isolated in the .github/workflows directory so .github/devcontainer is a non-conflicting path.

I can research and add some documentation as I find it, I'll include opinions/conventions documented both for and against this type of use case expansion.

Let me do that real quick.

Reply:
Here are my references.

* Example of expanded .github usage & hygiene: 
  * [github.com/microsoft/terminal/.github](https://github.com/microsoft/terminal/tree/main/.github)
  * [docs.github.com/en/actions/learn-github-actions](https://docs.github.com/en/actions/learn-github-actions/understanding-github-actions)
  * [opensource.creativecommons.org/contributing-code/github-repo-guidelines](https://opensource.creativecommons.org/contributing-code/github-repo-guidelines/)
  * [probot.github.io/docs/best-practices](https://probot.github.io/docs/best-practices/#include-dry-run-functionality)
  * [scroll to .github/scripts definition](https://aws.amazon.com/blogs/devops/integrating-with-github-actions-ci-cd-pipeline-to-deploy-a-web-app-to-amazon-ec2/)
  * [github.com/joelparkerhenderson/github-special-files-and-path](https://github.com/joelparkerhenderson/github-special-files-and-paths#introduction)
  * [.github/actions](wallis.dev/blog/composite-github-actions)

The only well defined argument alternative I've found so far is [dot-config.github.io](https://dot-config.github.io/) which argues in favor of a standard root .config directory. There is more conversation on the topic on [ycombinator.com](https://news.ycombinator.com/item?id=24066748) of course.

Reply:
The general conversation seems to indicate the purpose of .github dir being for any non-conflicting use-case that supports the regular maintenance and operation of the repository's github related integrations.

For this reason I believe .github/devcontainer is a reasonable github asset which upon careful research does not conflict with any other path by that name.

Reply:
Good enough for me, and seems like a thing we can always change if we need to. Thanks for looking into it!